plugin/src/main/java/org/opensearch/ml/action/upload_chunk/MLModelMetaUploader.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.action.upload_chunk;
+
+import static org.opensearch.ml.common.CommonValue.ML_MODEL_INDEX;
+import static org.opensearch.ml.plugin.MachineLearningPlugin.TASK_THREAD_POOL;
+
+import java.time.Instant;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.client.Client;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.MLModel;
+import org.opensearch.ml.common.model.MLModelState;
+import org.opensearch.ml.common.transport.upload_chunk.MLUploadModelMetaInput;
+import org.opensearch.ml.indices.MLIndicesHandler;
+import org.opensearch.threadpool.ThreadPool;
+
+@Log4j2
+public class MLModelMetaUploader {
+
+    private final MLIndicesHandler mlIndicesHandler;
+    private final ThreadPool threadPool;
+    private final Client client;
+
+    @Inject
+    public MLModelMetaUploader(MLIndicesHandler mlIndicesHandler, ThreadPool threadPool, Client client) {
+        this.mlIndicesHandler = mlIndicesHandler;
+        this.threadPool = threadPool;
+        this.client = client;
+    }
+
+    public void uploadModelMeta(MLUploadModelMetaInput mlUploadModelMetaInput, ActionListener<String> listener) {
+        threadPool.executor(TASK_THREAD_POOL).execute(() -> {
```

Comment:
Actualy model meta doc creation is light weight, we can run it directly, no need to put it into ML thread pool.

Reply:
Sure. Will make the changes.